src/marqo/tensor_search/add_docs.py

```diff
@@ -44,10 +49,16 @@ def threaded_download_images(allocated_docs: List[dict], image_repo: dict,
         metric_obj = RequestMetricsStore.for_request()
         RequestMetricsStore.set_in_request(metrics=metric_obj)
 
+    def is_non_tensor_field(f: str) -> bool:
```

Comment:
would it be better to have this function at the file level, so it's easier to make unit tests for?

Reply:
I agree, I prefer to avoid nested functions and to avoid impure functions (functions that mutate/depends on an external state) if possible, unless absolutely necessary.

If we decide to move the function to the file level we could also add some straightforward tests for it